src/_pytest/hookspec.py

```diff
@@ -303,8 +300,10 @@ def pytest_collect_file(
 
     .. versionchanged:: 7.0.0
         The ``file_path`` parameter was added as a :class:`pathlib.Path`
-        equivalent of the ``path`` parameter. The ``path`` parameter
-        has been deprecated.
+        equivalent of the ``path`` parameter.
+
+    .. versionchanged:: 8.0.0
+        The ``path`` parameter has been deprecated.
```

Comment:
Actually it was removed, not deprecated.

Reply:
Was it though? It worked in 8.0 but does not work in 8.1 :shrug:

Reply:
as per deprecation policy, the major release tirggers errors, the next minor version removes the code

the key issue here that we missed a case where warnings don't error

Reply:
@RonnyPfannschmidt I get it. What is confusing to me is why the docs says that it was removed in 8.0 rather than that it was removed in 8.1:

![image](https://github.com/pytest-dev/pytest/assets/5832902/8a2cff8b-7fec-4812-bf00-031dfd8601c2)

From a user's point of view it would be ideal to see "deprecated in 8.0, removed in 8.1", otherwise I see that it worked in 8.0 but does not work in 8.1 and I am just confused.

Reply:
the bug here is that it should have failed in 8.x, but due to a warning miss-config it did in fact not fail
the deprecation did happen - see #12068 

i vaguely recall the deprecation happened later in the 7.x series as to avoid having dozens of separate deprecation events